brat/Data/Hugr.hs

```diff
@@ -192,44 +238,25 @@ instance ToJSON node => ToJSON (FuncDefn node) where
                                     ,"signature" .= signature_
                                     ]
 
-data HugrConst
-  = HCInt Int
-  | HCFloat Double
-  | HCUnit
-  | HCFunction (Hugr Int)
- deriving (Eq, Show)
+data CustomConst where
+  CC :: forall a. (Eq a, Show a, ToJSON a) => String -> a -> CustomConst
```

Comment:
Is `KeyMap` the only valid type we can have for `a`?

Reply:
It might as well be specialised

Reply:
Actually, we can't have `KeyMap` here, only `KeyMap a`